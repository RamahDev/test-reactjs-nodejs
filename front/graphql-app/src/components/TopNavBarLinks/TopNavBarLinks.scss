.root{
    .MuiButton-root, .MuiFormLabel-root, .MuiInputBase-input, .MuiIconButton-root{
        color:#fff;
    }
    
}
.auto-complete .MuiInput-underline:after { 
    border-bottom : 2px solid rgba(255, 255, 255, 0.87);
}

.auto-complete .MuiInput-underline:before, .main-settings-container .MuiInput-underline:hover:not(.Mui-disabled):before{
    border-bottom : 1px solid #fff; 
}

.auto-complete .MuiOutlinedInput-notchedOutline, .main-settings-container .MuiOutlinedInput-root:hover .MuiOutlinedInput-notchedOutline{
    border-color : #fff;
}
.auto-complete .MuiOutlinedInput-root.Mui-focused .MuiOutlinedInput-notchedOutline{
    border-color : #fff;
}

.auto-complete .MuiOutlinedInput-notchedOutline, .main-settings-container .MuiOutlinedInput-root:hover .MuiOutlinedInput-notchedOutline{
    border-color : #fff;
}

.auto-complete .MuiInput-underline:hover:not(.Mui-disabled):before {
    border-bottom: 2px solid rgba(255, 255, 255, 0.87);
}

.auto-complete .MuiFormLabel-root.Mui-focused, .auto-complete .MuiFormLabel-root, .auto-complete .MuiInputBase-root, .auto-complete .MuiIconButton-root{
    color:#fff;
}