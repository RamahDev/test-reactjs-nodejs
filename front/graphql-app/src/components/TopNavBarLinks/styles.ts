import { makeStyles } from "@material-ui/core";

export const useStyles = makeStyles((theme: any) => ({
  root: {
    flexGrow: 1,
  },
  menuButton: {
    marginRight: theme.spacing(2),
  },
  title: {
    display: "flex",
    alignItems: "center",
  },
  autoComplete: {
    width: "150px",
    color: "white",
    textAlign: "center",
  },
  toolbar: {
    display: "flex",
    justifyContent: "space-between",
  },
  languageContainer: {
    display: "flex",
  },
  languageIcon: {
    marginRight: "1rem",
  },
}));
