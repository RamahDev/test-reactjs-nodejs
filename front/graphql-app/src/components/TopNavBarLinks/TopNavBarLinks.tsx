import {
  Button,
  makeStyles,
  TextField,
  Toolbar,
  Typography,
} from "@material-ui/core";
import { useHistory } from "react-router-dom";
import AppBar from "@material-ui/core/AppBar/AppBar";
import PublicIcon from "@material-ui/icons/Public";
import SportsEsportsIcon from "@material-ui/icons/SportsEsports";
import ViewStreamIcon from "@material-ui/icons/ViewStream";
import Autocomplete from "@material-ui/lab/Autocomplete";
import React, { useEffect, useState } from "react";
import { languages } from "../../data/languages";
import { useStyles } from "./styles";

import "./TopNavBarLinks.scss";

const TopNavBarLinks: React.FC = () => {
  const history = useHistory();
  const classes = useStyles();

  const [value, setValue] = useState<any>(languages[0]);

  const setLanguage = (language: any) => {
    if (language) {
      setValue(language);
      sessionStorage.setItem("language", language.value);
      window.location.reload();
    }
  };

  const getLanguage = (value: any) => {
    return languages.find((language) => language.value === value);
  };

  useEffect(() => {
    if (sessionStorage.getItem("language")) {
      setValue(getLanguage(sessionStorage.getItem("language")));
    } else {
      setValue(languages[0]);
      sessionStorage.setItem("language", languages[0].value);
    }
  }, []);

  const renderAutoComplete = () => {
    if (value) {
      return (
        <>
          <div className={classes.languageContainer}>
            <PublicIcon className={classes.languageIcon} />
            <Autocomplete
              className={`${classes.autoComplete} auto-complete`}
              id="size-small-standard"
              size="small"
              onChange={(event, newValue) => {
                setLanguage(newValue);
              }}
              options={languages}
              getOptionLabel={(option: any) => option.title}
              getOptionSelected={(option: any) => option.title}
              value={value}
              renderInput={(params: any) => (
                <TextField
                  {...params}
                  variant="standard"
                  placeholder="Langages"
                />
              )}
            />
          </div>
        </>
      );
    }
  };

  return (
    <div className={`${classes.root} root`}>
      <AppBar position="static">
        <Toolbar className={classes.toolbar}>
          <Typography variant="h6">Twitch Call React.js</Typography>
          <Typography variant="h6" className={classes.title}>
            <Button onClick={() => history.push("/")} color="default">
              <SportsEsportsIcon />
              Top Jeux
            </Button>
            <Button
              color="default"
              onClick={() => history.push("/streamers")}
            >
              <ViewStreamIcon />
              Streamers
            </Button>
            <Button
              color="default"
              onClick={() => history.push("/cron")}
            >
              <ViewStreamIcon />
              Cron
            </Button>
          </Typography>
          {renderAutoComplete()}
        </Toolbar>
      </AppBar>
    </div>
  );
};

export default TopNavBarLinks;
