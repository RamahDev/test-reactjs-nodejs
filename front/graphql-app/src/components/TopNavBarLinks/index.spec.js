import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import TopNavBarLinks from './TopNavBarLinks';

const toggleExpand = jest.fn();
const defaultProps = {
  title: 'awesome title',
  content: {},
  onClick: toggleExpand,
};

test('<TopNavBarLinks /> component renders correctly', () => {
  const { container } = render(<TopNavBarLinks {...defaultProps} />);
  expect(container).toMatchSnapshot();
});
