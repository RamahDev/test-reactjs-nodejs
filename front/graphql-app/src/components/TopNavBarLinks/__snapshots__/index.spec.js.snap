// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TopNavBarLinks /> component renders correctly 1`] = `
<div>
  <div
    class="makeStyles-root-1 root"
  >
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular makeStyles-toolbar-5 MuiToolbar-gutters"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6"
        >
          Twitch Call React.js
        </h6>
        <h6
          class="MuiTypography-root makeStyles-title-3 MuiTypography-h6"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M21.58 16.09l-1.09-7.66C20.21 6.46 18.52 5 16.53 5H7.47C5.48 5 3.79 6.46 3.51 8.43l-1.09 7.66C2.2 17.63 3.39 19 4.94 19c.68 0 1.32-.27 1.8-.75L9 16h6l2.25 2.25c.48.48 1.13.75 1.8.75 1.56 0 2.75-1.37 2.53-2.91zM11 11H9v2H8v-2H6v-1h2V8h1v2h2v1zm4-1c-.55 0-1-.45-1-1s.45-1 1-1 1 .45 1 1-.45 1-1 1zm2 3c-.55 0-1-.45-1-1s.45-1 1-1 1 .45 1 1-.45 1-1 1z"
                />
              </svg>
              Top Jeux
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M4 18h17v-6H4v6zM4 5v6h17V5H4z"
                />
              </svg>
              Streamers
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M4 18h17v-6H4v6zM4 5v6h17V5H4z"
                />
              </svg>
              Cron
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </h6>
        <div
          class="makeStyles-languageContainer-6"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root makeStyles-languageIcon-7"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-1 17.93c-3.95-.49-7-3.85-7-7.93 0-.62.08-1.21.21-1.79L9 15v1c0 1.1.9 2 2 2v1.93zm6.9-2.54c-.26-.81-1-1.39-1.9-1.39h-1v-3c0-.55-.45-1-1-1H8v-2h2c.55 0 1-.45 1-1V7h2c1.1 0 2-.9 2-2v-.41c2.93 1.19 5 4.06 5 7.41 0 2.08-.8 3.97-2.1 5.39z"
            />
          </svg>
          <div
            aria-expanded="false"
            class="MuiAutocomplete-root makeStyles-autoComplete-4 auto-complete MuiAutocomplete-hasClearIcon MuiAutocomplete-hasPopupIcon"
            role="combobox"
          >
            <div
              class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
            >
              <div
                class="MuiInputBase-root MuiInput-root MuiInput-underline MuiAutocomplete-inputRoot MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd MuiInputBase-marginDense MuiInput-marginDense"
              >
                <input
                  aria-autocomplete="list"
                  aria-invalid="false"
                  autocapitalize="none"
                  autocomplete="off"
                  class="MuiInputBase-input MuiInput-input MuiAutocomplete-input MuiAutocomplete-inputFocused MuiInputBase-inputAdornedEnd MuiInputBase-inputMarginDense MuiInput-inputMarginDense"
                  id="size-small-standard"
                  placeholder="Langages"
                  spellcheck="false"
                  type="text"
                  value="Tout"
                />
                <div
                  class="MuiAutocomplete-endAdornment"
                >
                  <button
                    aria-label="Clear"
                    class="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-clearIndicator MuiAutocomplete-clearIndicatorDirty"
                    tabindex="-1"
                    title="Clear"
                    type="button"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                  <button
                    aria-label="Open"
                    class="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-popupIndicator"
                    tabindex="-1"
                    title="Open"
                    type="button"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M7 10l5 5 5-5z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </header>
  </div>
</div>
`;
